src/facilities-context/transforms.tsx

```diff
@@ -0,0 +1,210 @@
+        value: version.staffPopulation as number,
+      })),
+    ["date"],
+  );
+
+  return getDateThresholdFunc(staffPopByDate);
+}
+
+function mergeModelVersions({
+  userVersions,
+  referenceFacility,
+}: Optional<MergedHistoryInputs, "referenceFacility">): ModelInputs[] {
+  const combinedVersions: ModelInputs[] = [...userVersions];
+  let { stateName, countyName } = combinedVersions.length
+    ? (last(combinedVersions) as ModelInputs)
+    : ({} as ModelInputs);
+
+  if (referenceFacility) {
+    const populationForDate = getPopulationFunc({
+      userVersions,
+      referenceFacility,
+    });
+
+    const capacityForDate = getCapacityFunc({
+      userVersions,
+      referenceFacility,
+    });
+
+    const staffPopByDate = getStaffPopulationFunc({ userVersions });
+    stateName = stateName || referenceFacility.stateName;
+    countyName = countyName || referenceFacility.countyName;
+
+    // fill any gaps with reference records:
+    // don't overwrite any user records with actual data
+    referenceFacility.covidCases.forEach(
+      ({ observedAt, popTestedPositive, staffTestedPositive }) => {
+        const existingRecord = combinedVersions.find((v) =>
+          isSameDay(v.observedAt, observedAt),
+        );
+
+        // although generally speaking reference data should NEVER replace
+        // existing user data, due to some quirks in our data modeling
+        // it is possible to save model versions that don't actually contain
+        // any case data; if that's the case we will actually want to use
+        // some reference data for that day
+        let shouldReplaceExisting = false;
+
+        if (existingRecord) {
+          if (hasCases(existingRecord)) return;
+
+          shouldReplaceExisting = true;
+        }
+
+        const newRecord = {
+          isReference: true,
+          observedAt,
+          ageUnknownCases: popTestedPositive,
+          ageUnknownPopulation: populationForDate(observedAt),
+          facilityCapacity: capacityForDate(observedAt),
+          staffCases: staffTestedPositive,
+          staffPopulation: staffPopByDate(observedAt),
+          stateName,
+          countyName,
+        };
+        if (shouldReplaceExisting && existingRecord) {
+          const i = combinedVersions.indexOf(existingRecord);
+          combinedVersions[i] = newRecord;
+        } else {
+          combinedVersions.push(newRecord);
+        }
+      },
+    );
+  }
+
+  return orderBy(
+    validateMergedModelVersions(combinedVersions),
+    ["observedAt"],
+    ["asc"],
+  );
+}
+
+export function mergeFacilityObjects({
+  userFacility,
+  referenceFacility,
+}: MergeProps): Facility {
+  const mergedFacility = { ...userFacility };
+
+  if (referenceFacility) {
+    // pull in additional facility metadata fields
+    const additionalMetadata = omit(referenceFacility, [
+      "id",
+      "stateName",
+      "facilityType",
+      "capacity",
+      "population",
+      "covidCases",
+    ]);
```

Comment:
hm yeah this is something i have changed my thinking on since writing this ... I think it would be better to use `pick` and enumerate the fields we want. I don't really see any benefit to just dumping arbitrary untyped metadata fields into this object; we can enumerate the ones we expect once we have an actual use for them. 

The omitted fields are excluded because we don't store these values this way on facilities; they are used elsewhere to populate the related fields in `modelInputs` and `modelVersions`

Reply:
after looking into it some more I think it's OK to leave this as is, but I will change the typedef and database function to be explicit and not include arbitrary fields.

I think the reason to still use `omit` here is that the goal, locally, is to exclude stuff we already have in user data and stuff that we know we are storing in a different form (the timeseries fields). If we added a second whitelist here we would have to keep it in sync with changes in the type definition, which seems like unnecessary redundancy. (at this point we don't expect any new timeseries fields to be added but there are various known static fields that we will get for free here once we start including them)